app/javascript/channels/testRunChannel.ts

```diff
@@ -0,0 +1,29 @@
+import consumer from '../utils/action-cable-consumer'
+import { camelizeKeys } from 'humps'
+import { Submission } from '../components/student/Editor'
+import { TestRun } from '../components/student/editor/TestRunSummary'
+import { typecheck } from '../utils/typecheck'
+
+export class TestRunChannel {
+  subscription: ActionCable.Channel
+
+  constructor(submission: Submission, received: (testRun: TestRun) => void) {
+    this.subscription = consumer.subscriptions.create(
+      {
+        channel: 'Submission::TestRunsChannel',
+        submission_uuid: submission.uuid,
+      },
+      {
+        received: (response: any) => {
+          const formattedResponse = camelizeKeys(response)
+
+          received(typecheck<TestRun>(formattedResponse, 'testRun'))
+        },
+      }
+    )
+  }
+
+  disconnect() {
+    this.subscription.unsubscribe()
+  }
+}
```

Comment:
Why the decision here to have a channel class as opposed to inlining like [we've done elsewhere](https://github.com/exercism/v3-website/blob/master/app/javascript/components/notifications/Icon.jsx#L15-L25)?

Reply:
I decided to do it this way as we want a separate layer to make sure that the message we receive back follows a type and that we change the keys to camel case. In the future, we might just have one channel class that does camelizing of keys, but I didn't want to generalize it yet.

Reply:
I quite like having the handling of the channel be in a separate class, as it feels like components should not necessarily be aware of the details of how to connect to a channel. I do agree that it is inconsistent with the other code, so maybe we should also introduce classes for the other channels?